src/gh/components/DF_visualization_settings/code.py

```diff
@@ -1,23 +1,203 @@
+        :param indx: the index of the input parameter
+        :param X_param_coord: the x coordinate of the input parameter
+        :param Y_param_coord: the y coordinate of the input parameter
+        :param X_offset: the offset of the slider from the input parameter
+    """
+    param = ghenv.Component.Params.Input[indx]
+    if param.SourceCount == 0:
+        slider = Grasshopper.Kernel.Special.GH_NumberSlider()
+        slider.NickName = nickname
+        slider.Description = "Set the value for the threshold"
+        slider.Slider.Minimum = System.Decimal(lower_bound)
+        slider.Slider.Maximum = System.Decimal(upper_bound)
+        slider.Slider.DecimalPlaces = 3
+        slider.Slider.SmallChange = System.Decimal(0.001)
+        slider.Slider.LargeChange = System.Decimal(0.01)
+        slider.Slider.Value = System.Decimal(default_value)
+        slider.CreateAttributes()
+        slider.Attributes.Pivot = System.Drawing.PointF(
+            X_param_coord - (slider.Attributes.Bounds.Width) - X_offset,
+            Y_param_coord - (slider.Attributes.Bounds.Height / 2 - 0.1)
+            )
+        slider.Attributes.ExpireLayout()
+        Grasshopper.Instances.ActiveCanvas.Document.AddObject(slider, False)
+        ghenv.Component.Params.Input[indx].AddSource(slider)
+
+def add_plane_object(self,
+    nickname: str,
+    indx: int,
+    X_param_coord: float,
+    Y_param_coord: float,
+    X_offset: int=75
+    ) -> None:
+    """
+        Adds a plane object to the component input
+
+        :param nickname: the nickname of the plane object
+        :param indx: the index of the input parameter
+        :param X_param_coord: the x coordinate of the input parameter
+        :param Y_param_coord: the y coordinate of the input parameter
+        :param X_offset: the offset of the plane object from the input parameter
+    """
+    param = ghenv.Component.Params.Input[indx]
+    if param.SourceCount == 0:
+        from Grasshopper import Instances
+        doc = Instances.ActiveCanvas.Document
+        if doc:
+            plane = Grasshopper.Kernel.Parameters.Param_Plane()
+            plane.NickName = nickname
+            plane.CreateAttributes()
+            plane.Attributes.Pivot = System.Drawing.PointF(
+                X_param_coord - (plane.Attributes.Bounds.Width) - X_offset,
+                Y_param_coord
+                )
+            plane.Attributes.ExpireLayout()
+            doc.AddObject(plane, False)
+            ghenv.Component.Params.Input[indx].AddSource(plane)
+        # Grasshopper.Instances.ActiveCanvas.Document.AddObject(plane, False)
+
+
+class DFVisualizationSettings(component):
+    def __init__(self):
+        self.poss_value_types = ["Dist", "RMSE", "MAX", "MIN", "STD"]
+        self.poss_palettes = ["Jet", "Rainbow", "RdPu", "Viridis"]
+        
+        ghenv.Component.ExpireSolution(True)
+        ghenv.Component.Attributes.PerformLayout()
+        params = getattr(ghenv.Component.Params, "Input")
```

Comment:
Hi @9and3 ! Just a small question: Do you know where we the documentation on the component class is ? I assume getattr() is inherited from there, but I couldn't find the McNeel or Compas documentation on that... Thanks !

Reply:
Because there is none! I can show you on Monday in the office! :)

Reply:
I am sorry, this can be actually simplified, you poked a got point @DamienGilliard :) I believe this would simply work with:
```py
params = ghenv.Component.Params.Input
```
I'll test it if it's the same, it should because basically `getattr()` is just a safer way to access attributes of a py object.